tests/infrastructure/golden_images/update_boot_source/test_boot_sources_vm.py

```diff
@@ -0,0 +1,75 @@
+import logging
+
+import pytest
+from ocp_resources.data_source import DataSource
+from ocp_resources.persistent_volume_claim import PersistentVolumeClaim
+
+from utilities.infra import assert_os_version_mismatch_in_vm, validate_os_info_vmi_vs_linux_os
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def boot_source_preference_from_data_source_dict(data_import_cron_matrix__function__):
+    return data_import_cron_matrix__function__[[*data_import_cron_matrix__function__][0]]["preference"]
+
+
+@pytest.fixture()
+def data_source_from_data_import_cron(
+    golden_images_namespace,
+    golden_images_persistent_volume_claims_scope_function,
+    golden_images_volume_snapshot_scope_function,
+    data_import_cron_matrix__function__,
+):
+    data_source = DataSource(name=[*data_import_cron_matrix__function__][0], namespace=golden_images_namespace.name)
+    if data_source.exists is None:
+        pytest.fail(f"Data Source {[*data_import_cron_matrix__function__][0]} is missing")
+    if data_source.source.kind == PersistentVolumeClaim.kind:
+        cluster_volumes = golden_images_persistent_volume_claims_scope_function
+    else:
+        cluster_volumes = golden_images_volume_snapshot_scope_function
+    assert any([data_source.name in volume.name for volume in cluster_volumes]), (
+        f"DataSource data_source {data_source.source.kind} {data_source.name} is missing"
+    )
+    return data_source
```

Comment:
Please check the DataSource condition is Ready=True
We should not check the PVC or a VolumeSnapshot unless we absolutely must